### PR TITLE
Fixed the Sharpness aware minimization docstring

### PR DESCRIPTION
**Pull request by hmludwig**
Fixed the docstring such that it gets displayed correctly (see  #826)

Screenshot 2024-02-29 at 12 22 58

### CODE REVIEW

File: optax/contrib/sam.py [hunk 1]
```diff
@@ -110,36 +110,37 @@ def sam(
...
-        On every call to `opt.update`, `grad_fn` will be called
-        `sync_period - 1` times, once for each adversarial update. It is usually
-        ok to use the same minibatch in each of those updates, as in the example
-        above, but you can use the second argument to select different batches
-        at each adversarial step:
-        ```
-        grad_fn = jax.grad(lambda params, i: loss(params, batches[i]))
-        ```
+  NOTE: 
+    When `opaque_mode=True`, the `update` function must be called with a
```
Comment (fabianp) [inline optax/contrib/sam.py:123]:
```suggestion
    When ``opaque_mode=True``, the ``update`` function must be called with a
```

File: optax/contrib/sam.py [hunk 2]
```diff
@@ -110,36 +110,37 @@ def sam(
...
+
+      opt = sam(outer_opt, adv_opt, opaque_mode=True)
+      ...
+      # In the training loop:
+      grad_fn = jax.grad(
+        lambda params, _: loss(params, batch, and_other_args))
+      updates, state = opt.update(updates, state, params, grad_fn=grad_fn)
+      params = optax.apply_updates(params, updates)
+
+    On every call to `opt.update`, `grad_fn` will be called
```
Comment (fabianp) [inline optax/contrib/sam.py:136]:
```suggestion
    On every call to ``opt.update``, ``grad_fn`` will be called
```

File: optax/contrib/sam.py [hunk 3]
```diff
@@ -110,36 +110,37 @@ def sam(
...
+    `sync_period - 1` times, once for each adversarial update. It is usually
```
Comment (fabianp) [inline optax/contrib/sam.py:137]:
```suggestion
    ``sync_period - 1`` times, once for each adversarial update. It is usually
```

**Review by fabianp — COMMENTED** [review]
thanks! Minor formatting suggestions :-)

**Comment by hmludwig** [comment]
Thanks for the suggestions. I adopted them :)

**Review by fabianp — APPROVED** [review]

**Comment by fabianp** [comment]
amazing, let's get this merged! 🚀

**Comment by hmludwig** [comment]
I accidentally pushed to the wrong branch. It seems my reset messed things up. Sorry for that.

**Comment by fabianp** [comment]
no worries, we can easily get the original code
